workflows/funcscan.nf

```diff
@@ -114,44 +117,57 @@ workflow FUNCSCAN {
         ch_versions = ch_versions.mix(SEQKIT_SEQ_LENGTH.out.versions)
     }
     else {
-        ch_input_for_annotation = ch_intermediate_input.fastas.map { meta, fasta, protein, gbk -> [meta, fasta] }
+        ch_input_for_annotation = ch_intermediate_input.fastas.map { meta, fasta, protein, gff, gbk  -> [meta, fasta] }
     }
 
     /*
         ANNOTATION
     */
 
     // Some tools require annotated FASTAs
-    if ((params.run_arg_screening && !params.arg_skip_deeparg) || params.run_amp_screening || params.run_bgc_screening) {
+    if ((params.run_arg_screening && !params.arg_skip_deeparg) || params.run_amp_screening || params.run_bgc_screening || params.run_cazyme_screening) {
         ANNOTATION(ch_input_for_annotation)
         ch_versions = ch_versions.mix(ANNOTATION.out.versions)
 
         ch_new_annotation = ch_input_for_annotation
             .join(ANNOTATION.out.faa)
+            .join(ANNOTATION.out.gff)
             .join(ANNOTATION.out.gbk)
     }
     else {
         ch_new_annotation = ch_intermediate_input.fastas
     }
 
     // Mix back the preannotated samples with the newly annotated ones
-    ch_prepped_input = ch_new_annotation
-        .filter { meta, fasta, faa, gbk -> meta.category != 'long' }
+    ch_new_annotation_short = ch_new_annotation
+        .filter { meta, fasta, faa, gff, gbk -> meta.category != 'long' }
+
+    // Add gff_type to meta for cazyme screening
+    if ((params.run_cazyme_screening && !params.cazyme_skip_dbcan && (!params.dbcan_skip_cgc || !params.dbcan_skip_substrate)) && params.annotation_tool in ['pyrodigal', 'prodigal', 'prokka', 'bakta']) {
+      ch_new_annotation_short.map { meta, fasta, faa, gff, gbk ->
+          def new_meta = meta + [gff_type: 'prodigal']  // Only Use 'prodigal' as dbcan does not distinguish 'pyrodigal' and 'prodigal' 
```

Comment:
But what if the annotation is from `prokka` or `bakta` 🤔 , this would be the wrong type of GFF, right?